numpy/add_newdocs.py

```diff
@@ -463,6 +463,46 @@ def luf(lamdaexpr, *args, **kwargs):
 
     """))
 
+add_newdoc('numpy.core', 'nested_iters',
+    """
+    Create a ordered sequence of iterators. They are nested so that iterating
+    one will reset all the iterators after it in the sequence.
```

Comment:
"advancing one" would be better than "iterating one".

Also, it doesn't just reset the state of the inner iterators - it changes the inner iterators to point at the new element in the outer iterator.

Worth adding a note that the iterators are returned outermost-first.